src/Select.purs

```diff
@@ -90,266 +77,304 @@ derive instance ordVisibility :: Ord Visibility
-      -- key events and expire their search after a set number of milliseconds.
-      _, _ -> pure unit
-
-  Highlight target -> do
-    st <- H.get
-    when (st.visibility == On) do
-      H.modify_ _ { highlightedIndex = Just $ getTargetIndex st target }
-
-  Select target mbEv -> do
-    for_ mbEv (H.liftEffect <<< preventDefault <<< ME.toEvent)
-    st <- H.get
-    when (st.visibility == On) case target of
-      Index ix -> handleEvent $ Selected ix
-      Next -> handleEvent $ Selected $ getTargetIndex st target
-      Prev -> handleEvent $ Selected $ getTargetIndex st target
-
-  ToggleClick ev -> do
-    H.liftEffect $ preventDefault $ ME.toEvent ev
-    st <- H.get
-    case st.visibility of
-      On -> do
-        handle $ Focus false
-        handle $ SetVisibility Off
-      Off -> do
-        handle $ Focus true
-        handle $ SetVisibility On
-
-  Focus shouldFocus -> do
-    inputElement <- H.getHTMLElementRef $ H.RefLabel "select-input"
-    for_ inputElement \el -> H.liftEffect case shouldFocus of
-      true -> HTMLElement.focus el
-      _ -> HTMLElement.blur el
-
-  Key ev -> do
-    void $ H.fork $ handle $ SetVisibility On
-    let preventIt = H.liftEffect $ preventDefault $ KE.toEvent ev
-    case KE.key ev of
-      x | x == "ArrowUp" || x == "Up" ->
-        preventIt *> handle (Highlight Prev)
-      x | x == "ArrowDown" || x == "Down" ->
-        preventIt *> handle (Highlight Next)
-      x | x == "Escape" || x == "Esc" -> do
-        inputElement <- H.getHTMLElementRef $ H.RefLabel "select-input"
-        preventIt
-        for_ inputElement (H.liftEffect <<< HTMLElement.blur)
-      "Enter" -> do
-        st <- H.get
-        preventIt
-        for_ st.highlightedIndex \ix ->
-          handle $ Select (Index ix) Nothing
-      otherKey -> pure unit
-
-  PreventClick ev ->
-    H.liftEffect $ preventDefault $ ME.toEvent ev
-
-  SetVisibility v -> do
-    st <- H.get
-    when (st.visibility /= v) do
-      H.modify_ _ { visibility = v, highlightedIndex = Just 0 }
-      handleEvent $ VisibilityChanged v
-
-  Action act -> handleAction' act
-
-  where
-  -- eta-expansion is necessary to avoid infinite recursion
-  handle act = handleAction handleAction' handleEvent act
-
-  getTargetIndex st = case _ of
-    Index i -> i
-    Prev -> case st.highlightedIndex of
-      Just i | i /= 0 -> i - 1
-      _ -> lastIndex st
-    Next -> case st.highlightedIndex of
-      Just i | i /= lastIndex st -> i + 1
-      _ -> 0
     where
-    -- we know that the getItemCount function will only touch user fields,
-    -- and that the state record contains *at least* the user fields, so
-    -- this saves us from a set of unnecessary record deletions / modifications
-    userState :: State st -> {| st }
-    userState = unsafeCoerce
-
-    lastIndex :: State st -> Int
-    lastIndex = (_ - 1) <<< st.getItemCount <<< userState
+      -- | An array of `IProps` with `ToggleProps`. It
```

Comment:
These comments aren't going to show up in documentation, so we ought to add this documentation elsewhere, perhaps in a table in the README or something. That can be done in a subsequent update, just noting it here.

Reply:
Yeah... I was wondering about that...

Reply:
So... What if we put it on top of the type declarations (e.g. `ToggleProps`).

Reply:
Or maybe on top of `SelectReturn`? Because more people are likely to hover over that...

Reply:
I think over the type declarations makes the most sense

Reply:
Done. I added `ContainerProps` even though it's not really needed so that we can be consistent.